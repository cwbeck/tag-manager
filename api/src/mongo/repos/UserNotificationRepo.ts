import Repo from '../abstractions/Repo';
import { injectable } from 'inversify';
import UserNotification from '../models/UserNotification';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class UserNotificationRepo extends Repo<UserNotification> {
    protected readonly EXPIRES_AFTER = 86400 * 90;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                _created_at: 1,
            },
            expireAfterSeconds: this.EXPIRES_AFTER,
        },
    ];
}
