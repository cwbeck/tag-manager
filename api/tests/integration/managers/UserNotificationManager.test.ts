import ManagerTestUtils from '../../utils/ManagerTestUtils';
import TestUtils from '../../utils/TestUtils';
import UserNotificationRepo from '../../../src/mongo/repos/UserNotificationRepo';
import Mock = jest.Mock;

describe('UserNotificationManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetch UserNotification through user me', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockAsLoggedUser();

        const mockSutModel = TestUtils.buildMockNotification();
        const mockSutRepo = {} as UserNotificationRepo;
        mockSutRepo.find = jest.fn().mockResolvedValue([mockSutModel]);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                me {
                    id
                    user_notifications {
                        id
                        type
                        is_viewed
                    }
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
