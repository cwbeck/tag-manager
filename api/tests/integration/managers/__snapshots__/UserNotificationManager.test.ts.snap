// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNotificationManager fetch UserNotification through user me 1`] = `
Object {
  "data": Object {
    "me": Object {
      "id": "5edf4c48e9615d9a1c66baaa",
      "user_notifications": Array [
        Object {
          "id": "5edf4c48e9615d9a1c66beee",
          "is_viewed": false,
          "type": "WELCOME",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
